src/main/java/facade/UserFacade.java

```diff
@@ -74,5 +136,11 @@ public void logout(String username, String cookie) throws InvalidUserCookieExcep
    *
    */
   public void updateUser(User user, String cookie) throws UserNotFoundException, InvalidUserCookieException {
+
+      LOGGER.info("UserFacade.updateUser");
+      if(!validateUserCookie(user.getUsername())){
+          throw new InvalidUserCookieException();
+      }
+      userDao.updateUser(user);
```

Comment:
We need to sanitize the input to ensure the user's win/loss stats are not changed.